muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -690,18 +705,19 @@ public Response listWorkspaces(
     torWorkspaceWithCourseCredits.setWorkspaces(workspaces);
     torWorkspaceWithCourseCredits.setCompletedCourseCredits(allCourseCredits);
     torWorkspaceWithCourseCredits.setMandatoryCourseCredits(mandatoryCourseCredits);
-    torWorkspaceWithCourseCredits.setShowCredits(false);
-
-    User user = userController.findUserByDataSourceAndIdentifier(userIdentifier.getDataSource(), userIdentifier.getIdentifier());
+    torWorkspaceWithCourseCredits.setShowCredits(showCredits);
     
-    String curriculumName = guiderController.getCurriculumName(user.getCurriculumIdentifier());
+    return Response.ok(torWorkspaceWithCourseCredits).build();
+  }
+  
+  private String getCurriculumName(SchoolDataIdentifier curriculumIdentifier){
+    Map<SchoolDataIdentifier, String> curriculumNames = new HashMap<>();
```

Comment:
This will always create a new **empty** Map for every call of `getCurriculumName` kind of defeating the purpose why we want to add them to a Map in the first place. 

Using a separate function for it is fine approach (avoids repeating the if-block everywhere the name is needed), but the Map has to come in as a parameter from the calling function i.e. `private String getCurriculumName(Map<SchoolDataIdentifier, String> curriculumNameCache, SchoolDataIdentifier curriculumIdentifier) {`. After all, we want that if in the loop there are two or more calls with the same curriculumIdentifier, we use the value from the Map instead of loading it through guiderController.

You can verify it works this way with debug before changing this.